driver/src/main/java/org/neo4j/driver/internal/InternalResultStream.java

```diff
@@ -28,26 +28,25 @@
 import org.neo4j.driver.internal.spi.Connection;
 import org.neo4j.driver.internal.spi.StreamCollector;
 import org.neo4j.driver.internal.summary.SummaryBuilder;
-import org.neo4j.driver.v1.Function;
-import org.neo4j.driver.v1.Notification;
-import org.neo4j.driver.v1.Plan;
-import org.neo4j.driver.v1.ProfiledPlan;
 import org.neo4j.driver.v1.Record;
-import org.neo4j.driver.v1.RecordAccessor;
-import org.neo4j.driver.v1.ResultCursor;
-import org.neo4j.driver.v1.ResultSummary;
+import org.neo4j.driver.v1.ResultStream;
 import org.neo4j.driver.v1.Statement;
-import org.neo4j.driver.v1.StatementType;
-import org.neo4j.driver.v1.UpdateStatistics;
-import org.neo4j.driver.v1.Value;
 import org.neo4j.driver.v1.exceptions.ClientException;
 import org.neo4j.driver.v1.exceptions.NoSuchRecordException;
+import org.neo4j.driver.v1.summary.Notification;
+import org.neo4j.driver.v1.summary.Plan;
+import org.neo4j.driver.v1.summary.ProfiledPlan;
+import org.neo4j.driver.v1.summary.ResultSummary;
+import org.neo4j.driver.v1.summary.StatementType;
+import org.neo4j.driver.v1.summary.UpdateStatistics;
+import org.neo4j.driver.v1.util.Function;
+import org.neo4j.driver.v1.util.Functions;
+import org.neo4j.driver.v1.value.Value;
 
 import static java.lang.String.format;
 import static java.util.Collections.emptyList;
-import static org.neo4j.driver.v1.Records.recordAsIs;
 
-public class InternalResultCursor extends InternalRecordAccessor implements ResultCursor
+public class InternalResultStream implements ResultStream
```

Comment:
Agree, @boggle noted the same thing. I just picked something :) @boggle suggested `RecordResult`. Ideas welcome! Maybe even just `Result`?